geomstats/backend/numpy/__init__.py

```diff
@@ -191,7 +212,7 @@ def eval(x):
 
 
 def ndim(x):
-    return x.ndim
+    return _np.array(x).ndim
```

Comment:
I understand that you added this to prevent the error `int object has no attributed 'ndim'`. However, 1) this changes the behavior of numpy, 2) now geomstats does not follow the behavior of numpy.

On 1): now if someone writes gs.ndim(3) in numpy backend, there will be no error: this is confusing for numpy users.
On 2): we decided to follow numpy's API everywhere, to have a global logic.

Therefore, I would suggest dealing with this error __outside of the backend__, so that numpy stays consistent with itself and geomstats' stays consistent with numpy's logic. Happy to talk more about it though!